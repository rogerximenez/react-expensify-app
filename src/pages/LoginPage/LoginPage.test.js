import React from "react";
import { shallow } from "enzyme";
import { LoginPage } from "./LoginPage";

let wrapper, startLogin;
beforeEach(() => {
  startLogin = jest.fn();
  wrapper = shallow(<LoginPage startLogin={startLogin} />)
})

test('should render the Login Page', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should startLogin on click', () => {
  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalled();
})