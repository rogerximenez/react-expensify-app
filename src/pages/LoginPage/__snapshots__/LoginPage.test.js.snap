// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Login Page 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Log In To Your Account
    </h1>
    <p
      className="secondary-text"
    >
      Get your expenses under control.
    </p>
    <button
      className="btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
